File: skyspace666_ui/src/services/PaymentMethodService.js
import http from "../http-common"; 

class PaymentMethodService {
  getAllPaymentMethods(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/paymentMethod/paymentMethods`, searchDTO);
  }

  get(paymentMethodId) {
    return this.getRequest(`/paymentMethod/${paymentMethodId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/paymentMethod?field=${matchData}`, null);
  }

  addPaymentMethod(data) {
    return http.post("/paymentMethod/addPaymentMethod", data);
  }

  update(data) {
  	return http.post("/paymentMethod/updatePaymentMethod", data);
  }
  
  uploadImage(data,paymentMethodId) {
  	return http.postForm("/paymentMethod/uploadImage/"+paymentMethodId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new PaymentMethodService();

File: src/main/java/com/skyspace666/service/PaymentMethodService.java
package com.skyspace666.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.skyspace666.domain.PaymentMethod;
import com.skyspace666.dto.PaymentMethodDTO;
import com.skyspace666.dto.PaymentMethodSearchDTO;
import com.skyspace666.dto.PaymentMethodPageDTO;
import com.skyspace666.dto.PaymentMethodConvertCriteriaDTO;
import com.skyspace666.service.GenericService;
import com.skyspace666.dto.common.RequestDTO;
import com.skyspace666.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface PaymentMethodService extends GenericService<PaymentMethod, Integer> {

	List<PaymentMethod> findAll();

	ResultDTO addPaymentMethod(PaymentMethodDTO paymentMethodDTO, RequestDTO requestDTO);

	ResultDTO updatePaymentMethod(PaymentMethodDTO paymentMethodDTO, RequestDTO requestDTO);

    Page<PaymentMethod> getAllPaymentMethods(Pageable pageable);

    Page<PaymentMethod> getAllPaymentMethods(Specification<PaymentMethod> spec, Pageable pageable);

	ResponseEntity<PaymentMethodPageDTO> getPaymentMethods(PaymentMethodSearchDTO paymentMethodSearchDTO);
	
	List<PaymentMethodDTO> convertPaymentMethodsToPaymentMethodDTOs(List<PaymentMethod> paymentMethods, PaymentMethodConvertCriteriaDTO convertCriteria);

	PaymentMethodDTO getPaymentMethodDTOById(Integer paymentMethodId);







}





